utils/funcexp/func_quote.cpp

```diff
@@ -45,15 +45,14 @@ std::string Func_quote::getStrVal(rowgroup::Row& row, FunctionParm& fp, bool& is
     return "NULL";
   }
 
-  if (str.empty())
-    return "NULL";
+  size_t strSize = strlen(str.c_str());
```

Comment:
You have a shared_ptr to string why do you need to run strlen here?

Reply:
Because std::string can contain a data with NUL ('\0') inside.